repos/system_upgrade/common/actors/checkleftoverpackages/libraries/checkleftoverpackages.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.common.rpms import get_installed_rpms, get_leapp_dep_packages, get_leapp_packages
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledUnsignedRPM, LeftoverPackages, RPM
+
+
+def process():
+    LEAPP_PACKAGES = get_leapp_packages(major_version=[get_source_major_version()])
+    LEAPP_DEP_PACKAGES = get_leapp_dep_packages(major_version=[get_source_major_version()])
+    installed_rpms = get_installed_rpms()
+
+    if not installed_rpms:
+        return
+
+    leftover_pkgs_to_remove = LeftoverPackages()
+    unsigned = [pkg.name for pkg in next(api.consume(InstalledUnsignedRPM), InstalledUnsignedRPM()).items]
+
+    for rpm in installed_rpms:
+        rpm = rpm.strip()
+        if not rpm:
+            continue
+        try:
+            name, version, release, epoch, packager, arch, pgpsig = rpm.split('|')
+        except ValueError:
+            api.current_logger().warning('Could not parse rpm: {}'.format(rpm))
```

Comment:
Maybe adding some info that the rpm in unexpected format will be skipped will save some time debuggin? Like `Could not parser rpm: {}, skipping`?

Reply:
I'd consider moving rpm construction to a helper and reusing it here and in `_get_leftover_packages`

Reply:
Definitely makes sense, so I created helper function

Reply:
Correction: As mentioned bellow in the conversation. This function won't be included as shared function (at least not now)